lib/log-stream.js

```diff
@@ -22,80 +24,81 @@ class LogStream extends Readable {
 
     this.options = _.defaults(options, {
       durationBetweenPolls: 1000,
-      timeoutBeforeFirstLogs: 300*1000
+      timeoutBeforeFirstLogs: 300 * 1000
     });
 
     this.eventBuffer = [];
     this.pending = false;
     this.logsReceived = false;
     this.streamEnded = false;
-    this.cloudwatchlogs = new AWS.CloudWatchLogs();
+    this.cloudwatchlogs = new CloudWatchLogs({ region: region });
     this.stopRequested = false;
   }
 
   fetchLogs(_cb) {
     this.pending = true;
 
-    var params = {
+    const params = {
       logGroupName: this.options.logGroup,
       logStreamName: this.options.logStream,
       startFromHead: true,
       nextToken: this.nextToken
     };
 
-    var next = (_err, _data) => {
+    let next = (_err, _data) => {
       setTimeout(this._read.bind(this), this.options.durationBetweenPolls);
     };
 
-    this.cloudwatchlogs.getLogEvents(params, (err, data) => {
```

Comment:
Much of this function moves into separate then/catch blocks of the SDK function call (the new SDK is promise based rather than callback based). The error handling thus gets moved far down.